pymc3/distributions/distribution.py

```diff
@@ -293,6 +307,13 @@ def dist(
         # Create the RV with a `size` right away.
         # This is not necessarily the final result.
         rv_out = cls.rv_op(*dist_params, size=create_size, **kwargs)
+        if initval is None and hasattr(rv_out.tag, "test_value"):
+            warnings.warn(
+                f"An initval=None was specified, but the {cls.rv_op.__class__.__name__} Op assigned a test value anyways."
+                f" This unexpected behaviour and should be investigated!",
+                RuntimeWarning,
+            )
```

Comment:
In a future where we are fully compatible with randomly initvals and move away from `test_value` we don't want things like that to go unnoticed, right?